components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -33,6 +36,29 @@ class GeckoEngineSession(
         geckoSession.trackingProtectionDelegate = createTrackingProtectionDelegate()
     }
 
+    /**
+     * See [EngineSession.settings]
+     */
+    override val settings: Settings = object : Settings, PrefsHelper.PrefHandlerBase() {
+        val prefs = mutableMapOf<String, Any>()
+        init { PrefsHelper.getPref(DOM_STORAGE_PREF, this) }
+
+        override var javascriptEnabled: Boolean
+            get() = runtime.settings.javaScriptEnabled
+            set(value) { runtime.settings.javaScriptEnabled = value }
+
+        override var domStorageEnabled: Boolean
+            get() = prefs[DOM_STORAGE_PREF] as? Boolean ?: true
+            set(value) {
+                PrefsHelper.setPref(DOM_STORAGE_PREF, value)
+                prefs[DOM_STORAGE_PREF] = value
+            }
```

Comment:
Oh, I didn't know that `PrefsHelper` is exposed. For other things we can use `GeckoSessionSettings` and `GeckoRuntimeSettings`. 

@snorp Is using PrefsHelper okay?

Reply:
@snorp Heh, it *is* part of the public API. Otherwise I wouldn't be able to use it :). But, point taken, I will remove this for now. The type shouldn't be public then, I guess.

Reply:
It's difficult to actually give it `package` protection currently because of how the code is organized. We'll get to it some day, I hope.

Only stuff in org.mozilla.geckoview should be considered public, with a couple caveats.